egs/wsj/s5/steps/nnet3/train_dnn.py

```diff
@@ -555,6 +578,17 @@ def Train(args, run_opts):
     # egs_dir will be updated if there is realignment
     cur_egs_dir=egs_dir
 
+    # If true, compute the dropout proportion for training.
+    # and write dropout proportion.
+    dp_prop = [1.0] * num_iters
+    if add_ephemeral_connection:
+      init_zero_dp_iter = 10 + num_hidden_layers * args.add_layers_period
```

Comment:
Sure, adding dropout_schedule  is good too! ComputeDropout computes dropout rate per iteration. 
"num_init_zero_dropout" is also important variable. The whole idea is that we should not do dropout in the beginning of training to let the network use this connection and then increase dropout gradually using "dropout_schedule"! 
We should do more experiments to find the optimum "dropout_schedule".